src/plugins/vis_types/timeseries/public/application/components/timeseries_visualization.tsx

```diff
@@ -150,46 +155,50 @@ function TimeseriesVisualization({
   const [firstSeries] =
     (isVisTableData(visData) ? visData.series : visData[model.id]?.series) ?? [];
 
-  if (VisComponent) {
+  if (!VisComponent || !palettesService || indexPattern === null) {
```

Comment:
nitpick: I would say, for consistency, to either do `palettesService == null || indexPattern == null` or use the exclamation mark for both `!palettesService || !indexPattern`.